OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -925,6 +925,9 @@ void IResolveOrder.ResolveOrder(Actor self, Order order)
 
 			if (order.OrderString == "Move")
 			{
+				if (!order.Target.IsValidFor(self))
+					return;
+
```

Comment:
Made a separate commit.

I had too add this check for AttackMove as the change in the bot code to use that instead of Attack was crashing, so decided to add all missing checks. As you say most of them aren't stricly required but seems better to have them everywhere than to not.